state/azure/cosmosdb/cosmosdb.go

```diff
@@ -80,31 +67,27 @@ type CosmosOperation struct {
 
 // CosmosItem is a wrapper around a CosmosDB document.
 type CosmosItem struct {
-	documentdb.Document
 	ID           string      `json:"id"`
 	Value        interface{} `json:"value"`
 	IsBinary     bool        `json:"isBinary"`
 	PartitionKey string      `json:"partitionKey"`
 	TTL          *int        `json:"ttl,omitempty"`
-}
-
-type storedProcedureDefinition struct {
-	ID   string `json:"id"`
-	Body string `json:"body"`
+	Etag         string
 }
 
 const (
-	storedProcedureName   = "__dapr_v2__"
-	versionSpName         = "__daprver__"
 	metadataPartitionKey  = "partitionKey"
 	unknownPartitionKey   = "__UNKNOWN__"
 	metadataTTLKey        = "ttlInSeconds"
 	statusTooManyRequests = "429" // RFC 6585, 4
 	statusNotFound        = "NotFound"
 )
 
+// Value used for timeout durations
+const timeoutValue = 20
+
 // NewCosmosDBStateStore returns a new CosmosDB state store.
-func NewCosmosDBStateStore(logger logger.Logger) state.Store {
+func NewCosmosDBStateStore(logger logger.Logger) *StateStore {
 	s := &StateStore{
 		features: []state.Feature{state.FeatureETag, state.FeatureTransactional, state.FeatureQueryAPI},
```

Comment:
Is there a reason why the features need to be stored as a property in the object? They could just be returned by the `Features()` method without being a property in the object.